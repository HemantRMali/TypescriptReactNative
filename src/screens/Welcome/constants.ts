/**
 * Filename: User actionType.js
 */

// Define your action types and import it in required reducers.
export const UserActions = {
  SAVE_USER: 'SAVE_USER',
};
