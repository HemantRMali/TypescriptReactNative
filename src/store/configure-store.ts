import {createStore, applyMiddleware} from 'redux';
import {createLogger} from 'redux-logger';
// Imports: Redux Root Reducer
import appReducer from '../reducers';

// Redux: Store
const store = createStore(appReducer, applyMiddleware(createLogger()));

// Exports
export default store;
