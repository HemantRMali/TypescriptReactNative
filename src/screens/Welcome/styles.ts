import {StyleSheet} from 'react-native';
import {colors} from '../../assets/colors';
import {DeviceInfo} from '../../constants';
export default StyleSheet.create({
  safeArea: {flex: 1},
  container: {
    flex: 1,
    backgroundColor: colors.rgba240,
  },
  textInputAndButtonContainer: {
    marginTop: DeviceInfo.height * 0.2,
    marginHorizontal: 30,
  },
});
