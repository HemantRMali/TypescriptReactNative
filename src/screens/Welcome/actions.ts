import {User} from '../../store/types';
import {UserActions} from './constants';
/**
 * This action creator is used store user details.
 */

interface saveUserDetailsAction {
  type: typeof UserActions.SAVE_USER;
  payload: User;
}

export type UserActionTypes = saveUserDetailsAction;

// TypeScript infers that this function is returning saveUserDetails
export function saveUserDetails(user: User): UserActionTypes {
  return {
    type: UserActions.SAVE_USER,
    payload: user,
  };
}
