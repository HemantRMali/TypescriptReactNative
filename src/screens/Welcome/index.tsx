import React, {useEffect} from 'react';
import {View, NativeModules, Alert} from 'react-native';
import DefaultButton from '../../components/DefaultButton';
import ScreenTitle from '../../components/ScreenTitle';
import CustomTextInput from '../../components/DefaultTextInput';
import {ButtonVariationsEnum} from '../../enums';
import {useDispatch} from 'react-redux';
import {saveUserDetails} from './actions';
import {Screens, TextInputLabels, ButtonTitles} from '../../constants';
import {useNavigation} from '@react-navigation/native';
import styles from './styles';

const {TargetOSManager} = NativeModules;

const checkTargetOS = async () => {
  const message = await TargetOSManager.isRunningOnDevice();
  Alert.alert(message);
};

const Welcome = () => {
  const navigation = useNavigation();

  const [userName, setUserName] = React.useState('');

  const dispatch = useDispatch();

  useEffect(() => {
    checkTargetOS();
  }, []);

  const onPress = () => {
    if (userName.trim().length > 0) {
      const user = {name: userName};
      dispatch(saveUserDetails(user));
      navigation.navigate(Screens.buttonVariations);
    }
  };

  return (
    <View style={styles.container}>
      <ScreenTitle title=" Hi There, Welcome..!" />
      <View style={styles.textInputAndButtonContainer}>
        <CustomTextInput
          label={TextInputLabels.enterFullName}
          value={userName}
          isSecure={false}
          onChangeText={(value) => setUserName(value)}
        />
        <DefaultButton
          variationType={ButtonVariationsEnum.primary}
          title={ButtonTitles.continue}
          onPress={() => onPress()}
        />
      </View>
    </View>
  );
};

export default Welcome;
