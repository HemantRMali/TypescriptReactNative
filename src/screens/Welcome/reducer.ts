/**
 * Filename: Welcome reducer.js
 *
 */

/**
 * Constants
 */
import {UserActions} from './constants';
import {UserState} from '../../store/types';
import {UserActionTypes} from './actions';

// Initial State

const initialState: UserState = {
  user: {name: ''},
};

// Redux: UserReducer
const UserReducer = (state = initialState, action: UserActionTypes) => {
  switch (action.type) {
    case UserActions.SAVE_USER:
      return {
        ...state,
        user: action.payload,
      };

    default: {
      return state;
    }
  }
};

// Exports
export default UserReducer;
