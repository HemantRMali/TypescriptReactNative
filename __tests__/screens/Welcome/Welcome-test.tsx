// __tests__/Welcome-test.js
import React from 'react';
import {Provider} from 'react-redux';
import Welcome from '../../../src/screens/Welcome';
import store from '../../../src/store/configure-store';
import {render, fireEvent} from '@testing-library/react-native';
import {Alert} from 'react-native';
describe('Welcome screen', () => {
  const {toJSON, getByTestId, getByPlaceholderText} = render(
    <Provider store={store}>
      <Welcome />
    </Provider>,
  );

  describe('Tapping on continue', () => {
    it('checks the user name', () => {
      fireEvent.changeText(getByPlaceholderText('Enter Full Name'), '');
      fireEvent.press(getByTestId('continueButton'));
      jest.spyOn(Alert, 'alert');

      fireEvent.changeText(getByTestId('userName'), 'Hemant Mali');
      fireEvent.press(getByTestId('continueButton'));

      expect(getByTestId('userName').props.value.length).toBeGreaterThanOrEqual(
        3,
      );
    });
  });
  expect(toJSON()).toMatchSnapshot();
});
