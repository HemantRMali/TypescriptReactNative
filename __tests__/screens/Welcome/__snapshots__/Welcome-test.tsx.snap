// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#595856",
          "fontSize": 40,
          "margin": 20,
        },
        undefined,
      ]
    }
  >
     Hi There, Welcome..!
  </Text>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 266.8,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#00000000",
          "borderColor": "#49658c",
          "borderRadius": 12,
          "borderWidth": 2,
          "color": "#49658c",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 30,
          "paddingBottom": 10,
          "paddingHorizontal": 11,
          "paddingTop": 10,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#49658c",
            "fontSize": 14,
            "left": 0,
            "paddingLeft": 10,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 14,
              },
            ],
            "zIndex": 3,
          }
        }
      >
        Enter Full Name
      </Text>
      <TextInput
        allowFontScaling={true}
        containerStyles={
          Object {
            "marginBottom": 30,
          }
        }
        label="Enter Full Name"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onSelectionChange={[Function]}
        onSubmitEditing={[Function]}
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "color": "#49658c",
            "flex": 1,
            "minHeight": 28,
            "paddingVertical": 0,
            "zIndex": 10,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "steelblue",
            "borderRadius": 5,
            "justifyContent": "center",
            "margin": 20,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 20,
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
  </View>
</View>
`;
